src/converters/degiroConverterV3.ts

```diff
@@ -62,90 +69,83 @@ export class DeGiroConverterV3 extends AbstractConverter {
         activities: []
       };
 
+      const isinHardcodeMapping = new Map<string, string>();
+      if (process.env.DEGIRO_V3_ISIN_HARDCODE_MAPPING) {
+        /* Sometimes there are cases where getting security gets a wrong result.
+         * For instance dividends for IE00B02KXK85 (ISHARES CHINA LARGE CAP UCITS ETF USD)
+         * which are in USD are detected as Invesco CurrencyShares Canadian Dollar Trust (FXC).
+         * So, if an imported file has first dividend payment the rest of the transactions
+         * are registered under wrong security. If first transaction is buy/sell, then it's all good
+         * since buy/sell txs are in EUR.
+         *
+         * This hack allows to buypass all internal heuristic and hardcode correct results.
+         * Format: IE00B02KXK85=FXAC.AS,foo=bar
+         */
+
+        if (process.env.DEGIRO_V3_ISIN_HARDCODE_MAPPING) {
+          const pairs = process.env.DEGIRO_V3_ISIN_HARDCODE_MAPPING.split(',');
+          pairs.forEach(p => {
+            const kv = p.split('=');
+            if (kv.length == 2 && kv[0] && kv[1]) {
+              isinHardcodeMapping.set(kv[0].trim().toLowerCase(), kv[1].trim().toLowerCase());
+            }
+          });
+        }
+      }
+
       // Populate the progress bar.
       const bar1 = this.progress.create(records.length, 0);
 
-      for (let idx = 0; idx < records.length; idx++) {
-        const record = records[idx];
+      // HashSet to skip processed records
+      const processedRecords = new Set<string>();
+
+      // Map plain objects to DeGiroRecord instances
+      const deGiroRecords = records.map(record => DeGiroRecord.fromPlainObject(record));
+
+      for (let idx = 0; idx < deGiroRecords.length; idx++) {
+        const record = deGiroRecords[idx];
 
         // Check if the record should be ignored. 
         if (this.isIgnoredRecord(record)) {
           bar1.increment();
           continue;
         }
 
-        // Look if the current record was already processed previously by checking the orderId.
-        // Not all exports provide an order ID, so check for a buy/sell marking in those cases.
-        // Dividend records never have an order ID, so check for a marking there.
-        // If a match was found, skip the record and move next.
-        if (result.activities.findIndex(a =>
-          a.comment !== "" &&
-          a.comment === record.orderId ||
-          a.comment.startsWith(`Buy ${record.isin} @ ${record.date}T`) ||
-          a.comment.startsWith(`Sell ${record.isin} @ ${record.date}T`) ||
-          a.comment.startsWith(`Dividend ${record.isin} @ ${record.date}T`)) > -1) {
-
-          bar1.increment();
-          continue;
+        // Check if the current record was already processed.
+        const recordHash = this.hashRecord(record);
```

Comment:
This might be a better solution. However I need to look into this better on my pc where I have a better overview of the whole change set.

Reply:
sure, let me know